src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/ElectricGoldPan.java

```diff
@@ -80,30 +87,24 @@ public ItemStack getProgressBar() {
 
     @Override
     protected MachineRecipe findNextRecipe(BlockMenu menu) {
-        if (!isOutputLimitOverriden() && !hasFreeSlot(menu)) {
+        if (!isOutputLimitOverridden() && !hasFreeSlot(menu)) {
             return null;
         }
 
         for (int slot : getInputSlots()) {
             ItemStack item = menu.getItemInSlot(slot);
+            ItemStack output = null;
 
-            if (SlimefunUtils.isItemSimilar(item, gravel, true, false)) {
-                ItemStack output = goldPan.getRandomOutput();
-                MachineRecipe recipe = new MachineRecipe(3 / getSpeed(), new ItemStack[] { gravel }, new ItemStack[] { output });
-
-                if (output.getType() != Material.AIR && menu.fits(output, getOutputSlots())) {
-                    menu.consumeItem(slot);
-                    return recipe;
-                }
-            } else if (SlimefunUtils.isItemSimilar(item, soulSand, true, false)) {
-                ItemStack output = netherGoldPan.getRandomOutput();
-                MachineRecipe recipe = new MachineRecipe(4 / getSpeed(), new ItemStack[] { soulSand }, new ItemStack[] { output });
-
-                if (output.getType() != Material.AIR && menu.fits(output, getOutputSlots())) {
-                    menu.consumeItem(slot);
-                    return recipe;
-                }
+            if (goldPan.isValidInput(item)) {
+                output = goldPan.getRandomOutput();
+            } else if (netherGoldPan.isValidInput(item)) {
+                output = netherGoldPan.getRandomOutput();
+            }
 
+            if (output != null && output.getType() != Material.AIR && menu.fits(output, getOutputSlots())) {
+                MachineRecipe recipe = new MachineRecipe(4 / getSpeed(), new ItemStack[] { item }, new ItemStack[] { output });
```

Comment:
just asking about this tho, normal goldpan was 3/speed and nethergoldpan was 4/speed. is this change intentional?
should this also updates to lores and stuff on recipe (if any), or will it be automatically updated?

Reply:
Oh good point, didn't catch that Ill change that rn